components/pd_client/src/client.rs

```diff
@@ -282,6 +283,109 @@ impl fmt::Debug for RpcClient {
+                                if c.has_error() {
+                                    error!(
+                                        "failed to load global config with key {:?}",
+                                        c.get_error()
+                                    );
+                                } else {
+                                    res.insert(c.get_name().to_owned(), c.get_value().to_owned());
+                                }
+                            }
+                            Ok(res)
+                        }
+                        Err(err) => Err(box_err!("{:?}", err)),
+                    }
+                }
+            }) as PdFuture<_>,
+            Err(err) => Box::pin(async move { Err(box_err!("{:?}", err)) }) as PdFuture<_>,
+        };
+        self.pd_client
+            .request(req, executor, LEADER_CHANGE_RETRY)
+            .execute()
+    }
+
+    fn store_global_config(&self, list: HashMap<String, String>) -> PdFuture<()> {
+        use kvproto::pdpb::GlobalConfigItem;
+        use kvproto::pdpb::StoreGlobalConfigRequest;
+        let mut req = StoreGlobalConfigRequest::new();
+        for (k, v) in list {
+            req.changes.push({
+                let mut item = GlobalConfigItem::new();
+                item.set_name(k);
+                item.set_value(v);
+                item
+            });
+        }
+        let executor = |client: &Client, req| match client
+            .inner
+            .rl()
+            .client_stub
+            .clone()
+            .store_global_config_async(&req)
+        {
+            Ok(grpc_response) => Box::pin(async move {
+                match grpc_response.await {
+                    Ok(grpc_response) => {
+                        if grpc_response.has_error() {
+                            error!(
+                                "error received in store global config {:?}",
+                                grpc_response.get_error()
+                            );
+                            Err(box_err!("{:?}", grpc_response.get_error()))
+                        } else {
+                            Ok(())
+                        }
+                    }
+                    Err(err) => Err(box_err!("{:?}", err)),
+                }
+            }) as PdFuture<_>,
+            Err(err) => Box::pin(async move { Err(box_err!("{:?}", err)) }) as PdFuture<_>,
+        };
+        self.pd_client
+            .request(req, executor, LEADER_CHANGE_RETRY)
+            .execute()
+    }
+
+    fn watch_global_config(
+        &self,
+    ) -> PdFuture<grpcio::ClientSStreamReceiver<pdpb::WatchGlobalConfigResponse>> {
+        use kvproto::pdpb::WatchGlobalConfigRequest;
+        let req = WatchGlobalConfigRequest::default();
+        let executor = |client: &Client, req| match client
+            .inner
+            .rl()
+            .client_stub
+            .clone()
+            .watch_global_config(&req)
```

Comment:
This method does not block. The outer `watch_global_config` needn't return a Future.